src/main/java/org/example/services/AuthService.java

```diff
@@ -0,0 +1,41 @@
+package org.example.services;
+
+import io.jsonwebtoken.Jwts;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+import org.example.daos.interfaces.IAuthDao;
+import org.example.exception.AuthDaoException;
+import org.example.exception.InvalidCredentialsException;
+import org.example.models.LoginRequest;
+import org.example.models.User;
+
+import java.security.Key;
+import java.util.Date;
+
+@Getter @Setter @AllArgsConstructor
+public class AuthService {
+    private final IAuthDao authDao;
+    private final Key key;
+
+    public String login(final LoginRequest loginRequest) throws
+            InvalidCredentialsException, AuthDaoException {
+        User user = authDao.getUser(loginRequest);
+        if (user.getUsername() == null || user.getPassword() == null) {
+            throw new InvalidCredentialsException();
+        }
+
+        return generateJwtToken(user);
+    }
+
+    private String generateJwtToken(final User user) {
+        return Jwts.builder()
+                .issuedAt(new Date(System.currentTimeMillis()))
+                .expiration(new Date(System.currentTimeMillis()))
```

Comment:
The issuedAt and expiration date and time are the same so this Jwt token would expire when it's created, making it unusable. So the expiration date should be a bit longer than the issuedAt date

Reply:
Fixed